apps/els_lsp/src/els_code_action_provider.erl

```diff
@@ -29,79 +29,133 @@ handle_request({document_codeaction, Params}, State) ->
 %% Internal Functions
 %%==============================================================================
 
-
 %% @doc Result: `(Command | CodeAction)[] | null'
 -spec code_actions(uri(), range(), code_action_context()) -> [map()].
-code_actions(Uri, _Range, Context) ->
-  #{ <<"diagnostics">> := Diagnostics } = Context,
-  Actions0 = [ make_code_action(Uri, D) || D <- Diagnostics],
-  Actions = lists:flatten(Actions0),
-  Actions.
-
-%% @doc Note: if the start and end line of the range are the same, the line
-%% is simply added.
--spec replace_lines_action(uri(), binary(), binary(), binary(), range())
-                          -> map().
-replace_lines_action(Uri, Title, Kind, Lines, Range) ->
-  #{ <<"start">> := #{ <<"character">> := _StartCol
-                     , <<"line">>      := StartLine }
-   , <<"end">>   := #{ <<"character">> := _EndCol
-                     , <<"line">>      := EndLine }
-   } = Range,
+code_actions(Uri, _Range, #{<<"diagnostics">> := Diagnostics}) ->
+  lists:flatten([make_code_action(Uri, D) || D <- Diagnostics]).
+
+-spec make_code_action(uri(), map()) -> [map()].
+make_code_action(Uri, #{<<"message">> := Message, <<"range">> := Range}) ->
+  make_code_action(
+    [ {"function (.*) is unused", fun action_export_function/3}
```

Comment:
Each diagnostic could return a number of code actions. For example, an unused function warning could result in the ability of exporting or removing a function. So, we could either pass a list of funs here or rename the fun into something like `unused_function_actions` and return there the list of options.